### ISSUE TITLE
[naming-convention] node env breaks global modifier on variable selector

### ISSUE THREAD

**Issue by burtek** [typescript-eslint/typescript-eslint#4781]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

**Repro**

```JSON
{
    "parser": "@typescript-eslint/parser",
    "plugins": ["@typescript-eslint"],
    "parserOptions": {
        "project": "./tsconfig.json"
    },
    "env": {
        "node": true
    },
    "rules": {
        "@typescript-eslint/naming-convention": ["error", {
            // default
            "selector": "default",
            "format": ["camelCase"]
        },
        {
            // const global variables can be UPPER_CASE
            "selector": "variable",
            "modifiers": ["const", "global"],
            "format": ["camelCase", "UPPER_CASE"]
        }]
    }
}
```

```TS
export const ABC_DEF = 123;
```

tsconfig as created by `tsc --init`.

The repro is also available as repository [burtek/naming-convention-global-bug-repro](https://github.com/burtek/naming-convention-global-bug-repro) - check README

**Expected Result**

`ABC_DEF` name should be allowed

**Actual Result**

```
Variable name `ABC_DEF` must match one of the following formats: camelCase eslint @typescript-eslint/naming-convention
```

![image](https://user-images.githubusercontent.com/2940958/161454211-46f50a33-7921-4a5b-87c5-307a8d65ae15.png)

**Additional Info**

The issue here is `node` env. When it's not set, rule works as expected. Adding `node: true` to `eslint`'s `env` config or `/* eslint-env node */` on top of `.ts` file makes the rule not work. The rbeaking modifier is `global` - removing it from modifiers list make the rule pass.

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `5.17.0` |
| `@typescript-eslint/parser`        | `5.17.0` |
| `TypeScript`                       | `4.6.3` |
| `ESLint`                           | `8.1.20` |
| `node`                             | `16.13.0` |

**Comment by burtek** [comment]
Just learnt `"globalReturn": false` needs to be added to `ecmaFeatures` in `parserOptions`. Works with that setting
